packages/sit-onyx/src/components/OnyxNavItem/OnyxNavItem.vue

```diff
@@ -0,0 +1,124 @@
+<script lang="ts" setup>
+import { computed, ref } from "vue";
+import type { OnyxNavItemProps } from "./types";
+import OnyxListbox from "../OnyxListbox/OnyxListbox.vue";
+import type { ListboxOption } from "../OnyxListbox/types";
+
+const props = withDefaults(defineProps<OnyxNavItemProps>(), {
+  active: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the navItem is clicked (via click or keyboard).
+   */
+  navigate: [href: string];
+}>();
+
+defineSlots<{
+  /**
+   * A default slot that can be used for OnyxBadge.
+   */
+  default(): unknown;
+}>();
+
+const listboxVisible = ref(false);
```

Comment:
I did it this way because there is a gap between the listbox and the navItem and while passing through this gap we are losing the listbox. You had a similar issue with the tooltip. How did you fix that?